packages/react/use-overlay/src/useOverlay.tsx

```diff
@@ -13,7 +13,7 @@ interface Options {
 export function useOverlay({ exitOnUnmount = true }: Options = {}) {
   const context = useContext(OverlayContext);
 
-  if (context == null) {
+  if (!context) {
```

Comment:
**Explicit Null or Undefined Check:** The original code explicitly checks for null or undefined, ensuring that the error is thrown only when context is truly not provided or initialized. This explicitness tends to make the code more readable and self-documenting since it conveys a clear intention that only null or undefined values are unacceptable.

**Avoiding False Positives:** JavaScript treats several values (like 0, "", false, and NaN) as falsy when evaluated in a boolean context. Switching to a general falsy check might introduce **unexpected scenarios where valid context values that are falsy** (e.g., a string with a length of 0 or the number 0) unintentionally trigger the error.

**Minimizing Risk**
Altering condition checks has the potential to introduce subtle bugs or alter the behavior of the code in certain edge cases. Without a compelling reason to adopt the broader falsy check, sticking to the explicit null or undefined check is a more conservative and safe approach.

Reply:
I understand. that's a very good reason